### ISSUE TITLE
Coffee not executing correctly

### ISSUE THREAD

**Issue by blainesch** [blainesch/pry.js#18]
This scenario fails:
~~~ cucumber
  Scenario: Setting the variable in coffeescript
    Given I have open the "fizzbuzz" example
    When I type in "mode"
    And I type in "i = 20"
    And I type in "i"
    Then The output should match "20"
~~~

The problem is when I type in `i = 20` it compiles to `var i; i = 10;`, typing the compiled js does the same thing.

**Comment by blainesch** [comment]
The best solution I can think of is to compile the snippet with the file and extract it.

**Comment by thisredone** [comment]
since CoffeeScript always places vars on a single line maybe we could just cut those out
I've also added auto setting of the coffee mode when the file has a .coffee extension #29

**Comment by afaur** [comment]
@thisredone You thinking on the single line vars thing is interesting. 

Something I think @blainesch may have forgotten to think about is that coffee-script also supports multiple assignment..
```
[a, b, c] = ['this', 'is', 'variables']
```
which turns into this js...
```
var a, b, c, ref;

ref = ['this', 'is', 'variables'], a = ref[0], b = ref[1], c = ref[2];
```
I am not sure how well that would work in that situation.

**Comment by blainesch** [comment]
I think removing the `var` statement would still work fine, assuming no assignments happen. I'm going to pull this down locally and play with it for a little bit before I merge it in. Maybe try a few versions of coffee to see if anything major changes

**Comment by blainesch** [comment]
Fixed by #33
